packages/aws-cdk/lib/api/bootstrap/bootstrap-environment.ts

```diff
@@ -145,12 +161,58 @@ export class Bootstrapper {
         CloudFormationExecutionPolicies: cloudFormationExecutionPolicies.join(','),
         Qualifier: params.qualifier,
         PublicAccessBlockConfiguration: params.publicAccessBlockConfiguration || params.publicAccessBlockConfiguration === undefined ? 'true' : 'false',
+        PermissionsBoundary: policyName,
       }, {
         ...options,
         terminationProtection: options.terminationProtection ?? current.terminationProtection,
       });
   }
 
+  private async getPolicyName(
+    environment: cxapi.Environment,
+    sdk: ISDK,
+    permissionsBoundary: string,
+    bootstrapTemplate: string,
+    params: BootstrappingParameters): Promise<string> {
+
+    if (permissionsBoundary === CDK_BOOTSTRAP_PERMISSIONS_BOUNDARY) {
+      const arn = await this.getExamplePermissionsBoundary(bootstrapTemplate, params.qualifier, environment.account, sdk);
+      const policyName = arn.split('/').pop();
+      permissionsBoundary = policyName ?? '';
+    }
+    this.validatePolicyName(permissionsBoundary);
+    return Promise.resolve(permissionsBoundary);
+  }
+
+  private validatePolicyName(permissionsBoundary: string) {
+    // https://docs.aws.amazon.com/IAM/latest/APIReference/API_CreatePolicy.html
+    const regexp: RegExp = /[\w+=,.@-]+/;
+    const matches = regexp.exec(permissionsBoundary);
+    if (!(matches && matches.length === 1 && matches[0] === permissionsBoundary)) {
+      throw new Error(`The permissions boundary name ${permissionsBoundary} does not match the IAM conventions.`);
+    }
+  }
+
+  private async getExamplePermissionsBoundary(bootstrapTemplate: any, qualifier: string | undefined, account: string, sdk: ISDK): Promise<string> {
+    const iam = sdk.iam();
+    // if no Qualifier is supplied, resort to the default one
+    const policyName = qualifier ? `cdk-${qualifier}-permissions-boundary` : 'cdk-hnb659fds-permissions-boundary';
+    const arn = `arn::iam::${account}:policy/${policyName}`;
+
+    let getPolicyResp = await iam.getPolicy({ PolicyArn: arn }).promise();
+    if (getPolicyResp.Policy) {
+      return arn;
+    }
+
+    const policyDoc = JSON.parse(serializeStructure(bootstrapTemplate, true)).CdkBoostrapPermissionsBoundaryPolicy;
+    const request = {
+      PolicyName: policyName,
+      PolicyDocument: JSON.stringify(policyDoc),
+    };
+    const createPolicyResponse = await iam.createPolicy(request).promise();
+    return createPolicyResponse.Policy?.Arn ?? '';
```

Comment:
I think if `iam.createPolicy` does not return an Arn we should throw an error
here rather than returning an empty string and having `validatePolicyName` throw
the error.